src/components/DateInput.jsx

```diff
@@ -172,7 +171,7 @@ export default class DateInput extends React.Component {
           placeholder={placeholder}
           autoComplete="off"
           disabled={disabled}
-          readOnly={readOnly || isTouch}
+          readOnly={readOnly}
```

Comment:
i wonder if we could make this semver-patch by changing this to `typeof readOnly === 'boolean' ? ‼readOnly : isTouch`, and making the `defaultProp` be `null`?